scripts/cxg_admin_scripts/migrate.py

```diff
@@ -201,3 +205,330 @@ def backfill_processing_status_for_datasets(ctx):
+                    }
+                    artifact_ids.append(record_artifact.id)
+                    artifacts.append(artifact)
+
+
+                dataset_version = {
+                    "version_id": dataset_version_id,
+                    "dataset_id": record_dataset.id,
+                    "collection_id": collection_id,
+                    "metadata": dataset_metadata,
+                    "artifacts": artifact_ids,
+                    "status": strip_prefixes_dict(status),
+                    "created_at": record_dataset.created_at,
+                }
+
+                dataset_ids.append(dataset_version_id)
+                datasets.append(dataset)
+                dataset_versions.append(dataset_version)
+            
+            version = {
+                "version_id": version_id,
+                "collection_id": collection_id,
+                "metadata": metadata,
+                "owner": record.owner,
+                "publisher_metadata": record.publisher_metadata,
+                "published_at": record.published_at,
+                "datasets": dataset_ids,
+                "created_at": record.created_at,
+            }
+
+            collection_versions.append(version)
+
+    import json
+    with open("migration/collections.json", "w") as f:
+        json.dump(collections, f, default=str)
+
+    with open("migration/collection_versions.json", "w") as f:
+        json.dump(collection_versions, f, default=str)
+
+    with open("migration/datasets.json", "w") as f:
+        json.dump(datasets, f, default=str)
+
+    with open("migration/dataset_versions.json", "w") as f:
+        json.dump(dataset_versions, f, default=str)
+
+    with open("migration/dataset_artifacts.json", "w") as f:
+        json.dump(artifacts, f, default=str)
+
+def migrate_redesign_debug(ctx):
+    import json
+    with open("migration/collections.json", "r") as f:
+        collections = json.load(f)
+
+    with open("migration/collection_versions.json", "r") as f:
+        collection_versions = json.load(f)
+
+    with open("migration/datasets.json", "r") as f:
+        datasets = json.load(f)
+
+    with open("migration/dataset_versions.json", "r") as f:
+        dataset_versions = json.load(f)
+
+    with open("migration/dataset_artifacts.json", "r") as f:
+        artifacts = json.load(f)
+
+    from backend.layers.common.entities import CollectionMetadata, DatasetStatus, DatasetMetadata
+
+    # for collection in collection_versions:
+    #     print(collection["metadata"])
+    #     CollectionMetadata.from_json(json.dumps(collection["metadata"]))
+
+    # for dataset in dataset_versions:
+    #     if not dataset["status"]:
+    #         print("skipping")
+    #         print(dataset)
+    #         continue
+    #     DatasetStatus.from_json(json.dumps(dataset["status"]))
+
+    for dataset in dataset_versions:
+        if dataset["metadata"]["schema_version"] not in  ["3.0.0", "2.0.0"]:
```

Comment:
do we still want 2.0.0 datasets?

Reply:
It's a good question. They're probably all tombstones, but there is no harm in migrating them though.

Reply:
They should not exist in prod, so I wouldn't worry.